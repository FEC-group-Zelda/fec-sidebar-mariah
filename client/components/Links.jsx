import React from 'react';
import Link from './Link.jsx';
import styled from 'styled-components';

import { FaInstagram, FaTwitter, FaFacebookF, FaConnectdevelop} from 'react-icons/fa';
// Website names

var array = ['Instagram', 'Twitter', 'Facebook', 'Website'];

//put social media icons here
var iconArray = ['➜', '♖', '∮', '◐' ];

const Ul = styled.ul`
  margin-block-start: 1em;
  margin-block-end: 1em;
  margin-inline-start: 0px;
  margin-inline-end: 0px;
  padding-inline-start: 40px;
  list-style: none;
  margin: 0;
`;


const Links = ({links}) => (
  <Ul>
    {links.map((link, index) =>

      <Link
        link={link}
        linkName={array[index]}
        icon={iconArray[index]}
      />
    )}
  </Ul>
);

export default Links;