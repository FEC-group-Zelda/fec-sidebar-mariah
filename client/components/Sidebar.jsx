import React from 'react';
import ReactDOM from 'react-dom';
import axios from 'axios';
import StatCounts from './StatCounts.jsx';
import Links from './Links.jsx';
import About from './About.jsx';


class Sidebar extends React.Component {

  constructor(props) {
    super(props);

    this.state = {
      links: '',
      about: '',
      follower_count: null,
      following_count: null,
      track_count: null,
    };
  }

  // send get req before component renders.
  componentWillMount() {
    // get req here
    var that = this;
    var name = 'Dedrick.Hauck'

    axios.get(`/artist/?name=${name}`)
      .then(function(artist) {
        that.setState(artist.data);
        console.log(JSON.stringify(artist.data));
      })
      .catch(function(err) {
        console.log(err);
      });
  }

  // on click function to be added when I have time


  render() {
    return (
      <div id='sidebar-right' style={{top: '401px'}}>

        <article id="stats">
          <table id="links">
            <tbody>
              <tr>

                <StatCounts statTitle='followers' statNum={this.state.follower_count} />
                <StatCounts statTitle='following' statNum={this.state.following_count} />
                <StatCounts statTitle='tracks' statNum={this.state.track_count} />
              </tr>

            </tbody>
          </table>

          <div id="about">
            <About about={this.state.about}/>
          </div>

        </article>
{/*

        <article>
          <Links links={this.state.links}/>
        </article> */}
      </div>
    );
  }
}



export default Sidebar;