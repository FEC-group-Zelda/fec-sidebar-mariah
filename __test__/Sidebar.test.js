//const app = require('../app.js');
import React from 'react';
import Sidebar from '../client/components/Sidebar.jsx';
import Links from '../client/components/Links.jsx';
import renderer from 'react-test-renderer';

import {configure, shallow, mount, render} from 'enzyme';
import Adapter from 'enzyme-adapter-react-15.4';

configure({adapter: new Adapter()});


// Test add new artist works
test('Four is not divisible by three', () => {
  expect( 4 % 3 !== 0).toBe(true);
});

// Test get requests from App
describe('Testing get requests from Sidebar App', () => {
  beforeall(() => {
    global.fetch = jest.fn();
    // global or window, whichever one works
  });

  let wrapper;

  beforeEach(() => {
    wrapper = shallow(<Sidebar />, {disableLifecycleMethods: true});
  });

  afterEach(() => {
    wrapper.unmount();
  });

  // Apply spy to componentwillmount

  const spyMount = jest.spyOn(Sidebar.prototype, 'componentWillMount');

  // // More complicated stuff for testing body of request.
  // fetch.mockImplementation(() => {
  //   return Promise.resolve({
  //     status: 200,
  //     json: () => {
  //       return Promise.resolve({

  //         artist: {
  //           info: 'placeholder'
  //         },
  //         likedsongs:
  //           ['song1', 'song2', 'song3']

  //       });
  //     }
  //   });
  // });

  // Call componentWillMount
  const didMount = wrapper.instance().componentWillMount;

  expect(spyMount).toHaveBeenCalled();


});


// Snapshot test for sidebar rendering.
test('Components render on mount', () => {
  const sidebar = renderer.create(<Sidebar />);
  var component = sidebar.toJSON();
  expect(sidebar).toMatchSnapshot();
});

// Displays the correct amount of links
test('Links render the correct amount', () => {

  const links = renderer.create(<Links links='onelink twolink threelink'/>);
  var test = links.toJSON();
  //Count amount of elements have id "sociallinks"
  //Compare with expected.
  expect(test).toMatchSnapshot();
});


// Hover over artist name test / Work in progress test.
test('Displays artist info over artist name hover', () => {
  const artistName = shallow(<LikedSong artist_name='TEST'/>);

  artistName.find('input').simulate('change');

  expect(artistName.text()).toEqual('hover');
});
