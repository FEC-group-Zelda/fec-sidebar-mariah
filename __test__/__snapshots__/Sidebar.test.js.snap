// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Components render on mount 1`] = `
<div
  id="sidebar-right"
  style={
    Object {
      "top": "401px",
    }
  }
>
  <article
    id="stats"
  >
    <table
      id="links"
    >
      <tbody>
        <tr>
          <td
            className="stat-column"
          >
            <h3
              class="stat-title"
            >
              followers
            </h3>
            <div
              class="stat-num"
            />
          </td>
          <td
            className="stat-column"
          >
            <h3
              class="stat-title"
            >
              following
            </h3>
            <div
              class="stat-num"
            />
          </td>
          <td
            className="stat-column"
          >
            <h3
              class="stat-title"
            >
              tracks
            </h3>
            <div
              class="stat-num"
            />
          </td>
        </tr>
      </tbody>
    </table>
  </article>
  <article>
    <div
      id="about"
    >
      <div
        id="collapsed"
      >
        <div
          id="wrapper"
          style={
            Object {
              "maxHeight": "85px",
            }
          }
        >
          <div
            id="truncated-description"
          >
            
          </div>
        </div>
      </div>
    </div>
  </article>
  <article>
    <div
      id="weblinks"
    >
      <ul />
    </div>
  </article>
  <article>
    <div
      id="likedsongs"
    >
      <div>
        <span
          className="LIKEDSONGCLASS"
        >
          <span>
             â™¡
            <h3
              className="LIKEDSONGSNUMBER"
            >
              0
               likes 
            </h3>
          </span>
        </span>
        <ul />
      </div>
    </div>
  </article>
</div>
`;
